### PR TITLE
Require current password when edit account

### PR DESCRIPTION
**Pull request by kimquy**
Improve this https://github.com/smpallen99/coherence/issues/107

### CODE REVIEW

File: lib/coherence/config.ex [hunk 1]
```diff
@@ -78,6 +78,7 @@ defmodule Coherence.Config do
     {:delete_login, :delete_login},
     {:opts, []},
     {:assigns_key, :current_user},
+    {:require_current_password, true},
```
Comment (smpallen99) [inline lib/coherence/config.ex]: I'm glad you made this configurable. Please update the docs above also.

File: lib/coherence/schema.ex [hunk 2]
```diff
@@ -151,7 +151,7 @@ defmodule Coherence.Schema do
         Returns a changeset ready for Repo.update
         """
         def unlock(user) do
-          Config.user_schema.changeset(user, %{locked_at: nil, unlock_token: nil, failed_attempts: 0})
+          Config.user_schema.changeset(user, %{locked_at: nil, unlock_token: nil, failed_attempts: 0, current_password: user.password})
```
Comment (smpallen99) [inline lib/coherence/schema.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: lib/coherence/schema.ex [hunk 3]
```diff
@@ -185,7 +185,7 @@ defmodule Coherence.Schema do
         Returns a changeset ready for Repo.update
         """
         def lock(user, locked_at \\ Ecto.DateTime.utc) do
-          Config.user_schema.changeset(user, %{locked_at: locked_at})
+          Config.user_schema.changeset(user, %{locked_at: locked_at, current_password: user.password})
```
Comment (smpallen99) [inline lib/coherence/schema.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: lib/coherence/schema.ex [hunk 4]
```diff
@@ -202,7 +202,7 @@ defmodule Coherence.Schema do
 
         def lock!(user, locked_at \\ Ecto.DateTime.utc) do
           IO.warn "#{inspect Config.user_schema}.lock!/1 has been deprecated. Please use Coherence.ControllerHelpers.lock!/1 instead."
-          changeset = Config.user_schema.changeset(user, %{locked_at: locked_at})
+          changeset = Config.user_schema.changeset(user, %{locked_at: locked_at, current_password: user.password})
```
Comment (smpallen99) [inline lib/coherence/schema.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: priv/templates/coherence.install/templates/coherence/password/edit.html.eex [hunk 5]
```diff
@@ -7,6 +7,12 @@
   <%%= hidden_input f, :reset_password_token %>
 
   <div class="form-group">
+    <%%= required_label f, :current_password, class: "control-label" %>
```
Comment (smpallen99) [inline priv/templates/coherence.install/templates/coherence/password/edit.html.eex]: This should only be displayed if the config is enabled

File: priv/templates/coherence.install/templates/coherence/registration/form.html.eex [hunk 6]
```diff
@@ -26,6 +26,14 @@
     <%%= error_tag f, :email %>
   </div>
 
+  <%%= if not is_nil(@changset.data.id) do %>
```
Comment (smpallen99) [inline priv/templates/coherence.install/templates/coherence/registration/form.html.eex]: This should only be displayed if the config is enabled

File: web/models/confirmable.ex [hunk 7]
```diff
@@ -57,7 +57,7 @@ defmodule Coherence.Schema.Confirmable do
         Returns a changeset ready for Repo.update
         """
         def confirm(user) do
-          Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+          Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```
Comment (smpallen99) [inline web/models/confirmable.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: web/models/confirmable.ex [hunk 8]
```diff
@@ -69,7 +69,7 @@ defmodule Coherence.Schema.Confirmable do
         """
         def confirm!(user) do
           IO.warn "#{inspect Config.user_schema}.confirm!/1 has been deprecated. Please use Coherence.ControllerHelpers.confirm!/1 instead."
-          changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+          changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```
Comment (smpallen99) [inline web/models/confirmable.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: web/models/confirmable.ex [hunk 9]
```diff
@@ -90,7 +90,7 @@ defmodule Coherence.Schema.Confirmable do
   deprecated! Please use Coherence.ControllerHelpers.unlock!/1.
   """
   def confirm(user) do
-    Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+    Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```
Comment (smpallen99) [inline web/models/confirmable.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

File: web/models/confirmable.ex [hunk 10]
```diff
@@ -101,7 +101,7 @@ defmodule Coherence.Schema.Confirmable do
   deprecated! Please use Coherence.ControllerHelpers.unlock!/1.
   """
   def confirm!(user) do
-    changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+    changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```
Comment (smpallen99) [inline web/models/confirmable.ex]: The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.

**Comment by kimquy** [comment]
Update readme

screen shot 2016-12-08 at 5 06 27 pm

**Comment by smpallen99** [comment]
Thanks!!

**Pull request merged**